code/ModelAdmin.php

```diff
@@ -321,19 +321,6 @@ protected function getGridFieldConfig(): GridFieldConfig
             ->addComponent($exportButton)
             ->addComponents(Injector::inst()->createWithArgs(GridFieldPrintButton::class, ['buttons-before-left']));
 
-        // Remove default and add our own filter header with extension points
-        // to retain API until deprecation in 5.0
-        $config->removeComponentsByType(GridFieldFilterHeader::class);
-        $config->addComponent(Injector::inst()->createWithArgs(GridFieldFilterHeader::class, [
-            false,
-            function ($context) {
-                $this->extend('updateSearchContext', $context);
-            },
-            function ($form) {
-                $this->extend('updateSearchForm', $form);
-            }
-        ]));
```

Comment:
We don't really have a way to deprecate these per-se, but they were removed in other admin sections already so it seems sensible to remove them from here as well.

Developers who need to update these can either:
1. update these in their own subclass of `ModelAdmin`
2. update these in their `DataObject` subclass (which is the preferred way)

Reply:
That said, I don't feel strongly about this - if there's any push back at all I'm happy to just keep these and remove the comment that suggests they should be deprecated.